Google.PowerShell/CloudResourceManager/IamPolicyCmdlets.cs

```diff
@@ -43,4 +49,266 @@ protected override void ProcessRecord()
+    public class AddGcIamPolicyBindingCmdlet : CloudResourceManagerCmdlet, IDynamicParameters
+    {
+        private class ParameterSetNames
+        {
+            public const string User = nameof(User);
+            public const string ServiceAccount = nameof(ServiceAccount);
+            public const string Group = nameof(Group);
+            public const string Domain = nameof(Domain);
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to set the IAM Policy Bindings. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public override string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Email address that represents a specific Google account.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.User)]
+        [ValidateNotNullOrEmpty]
+        public string User { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Email address that represents a service account.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ServiceAccount)]
+        [ValidateNotNullOrEmpty]
+        public string ServiceAccount { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Email address that represents a Google group.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.Group)]
+        [ValidateNotNullOrEmpty]
+        public string Group { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// A Google Apps domain name that represents all the users of that domain.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.Domain)]
+        [ValidateNotNullOrEmpty]
+        public string Domain { get; set; }
+
+        /// <summary>
+        /// This dynamic parameter dictionary is used by PowerShell to generate parameters dynamically.
+        /// </summary>
+        private RuntimeDefinedParameterDictionary _dynamicParameters;
+
+        /// <summary>
+        /// Dictionary of roles with key as project and value as the roles available in the project.
+        /// This dictionary is used for caching the various roles available in a project.
+        /// </summary>
+        private static ConcurrentDictionary<string, string[]> s_rolesDictionary =
+            new ConcurrentDictionary<string, string[]>();
+
+        private IamService iamService = new IamService(GetBaseClientServiceInitializer());
+
+        public object GetDynamicParameters()
```

Comment:
This is probably due to my lack of PS knowledge but what is this for? It seems to have a lot of specific processing

Should this be common code?